src/components/fields/ObjectField.js

```diff
@@ -615,59 +633,73 @@ class ObjectField extends Component {
       onJsonChange: this.onJsonChange,
       formJson: this.state.formJson,
       formJsonError: this.state.formJsonError,
-      properties: orderedProperties.map(name => {
-        if (
-          schema.discriminator &&
-          name === schema.discriminator &&
-          schema.discriminatorValue
-        ) {
-          discriminatorObj.name = name;
-          discriminatorObj.value = schema.discriminatorValue;
-        }
-
-        return {
-          content: (
-            <SchemaField
-              key={name}
-              name={name}
-              isEven={this.getIsEven()}
-              depth={this.getDepth()}
-              expandAll={this.state.expandAll}
-              expandAllLevel={this.state.expandAllLevel}
-              required={this.isRequired(name)}
-              schema={templateProps.schema.properties[name]}
-              uiSchema={templateProps.uiSchema[name]}
-              errorSchema={templateProps.errorSchema[name]}
-              idSchema={templateProps.idSchema[name]}
-              formData={templateProps.formData[name]}
-              onChange={this.onPropertyChange(name)}
-              onBlur={templateProps.onBlur}
-              onFocus={templateProps.onFocus}
-              registry={templateProps.registry}
-              disabled={templateProps.disabled || this.shouldDisable()}
-              readonly={templateProps.readonly}
-              disableFormJsonEdit={templateProps.disableFormJsonEdit}
-              typeCombinatorTypes={typeCombinatorTypes}
-              discriminatorObj={discriminatorObj}
-              // flag for direct circular reference in the objects
-              directCircularRef={
-                templateProps.schema.dataTypeLink ===
-                templateProps.schema.properties[name].dataTypeLink
-              }
-            />
-          ),
-          name,
-          readonly: templateProps.readonly,
-          disabled: templateProps.disabled || this.shouldDisable(),
-          required: templateProps.required,
-        };
-      }),
+      properties: checkSchemaForReadOnly(schema)
```

Comment:
function name is not clear, it creates confusion whether it checks schema for one time or it will check readonly for all properties